### PR TITLE
Fix logout when using URL with port

### PR DESCRIPTION
**Pull request by JohnAZoidberg**
The following code will fail:

```py
ilo_url = "https://192.168.100.207:443"
client = RedfishClient(base_url=ilo_url, username=username, password=password)
client.login()
client.logout()
```

with this error:

```
  File "/home/zoid/cloudhome/hpe/taiwan/venv-GNext/lib/python3.9/site-packages/redfish/rest/v1.py", line 356, in logout
    resp = self.delete(self.session_location)
  File "/home/zoid/cloudhome/hpe/taiwan/venv-GNext/lib/python3.9/site-packages/redfish/rest/v1.py", line 189, in delete
    return self.connection.rest_request(path, method='DELETE',
  File "/home/zoid/cloudhome/hpe/taiwan/venv-GNext/lib/python3.9/site-packages/redfish/rest/connections.py", line 257, in rest_request
    resp = self._conn(method, reqfullpath, **request_args)
  File "/home/zoid/cloudhome/hpe/taiwan/venv-GNext/lib/python3.9/site-packages/urllib3/request.py", line 74, in request
    return self.request_encode_url(
  File "/home/zoid/cloudhome/hpe/taiwan/venv-GNext/lib/python3.9/site-packages/urllib3/request.py", line 96, in request_encode_url
    return self.urlopen(method, url, **extra_kw)
  File "/home/zoid/cloudhome/hpe/taiwan/venv-GNext/lib/python3.9/site-packages/urllib3/poolmanager.py", line 362, in urlopen
    u = parse_url(url)
  File "/home/zoid/cloudhome/hpe/taiwan/venv-GNext/lib/python3.9/site-packages/urllib3/util/url.py", line 394, in parse_url
    return six.raise_from(LocationParseError(source_url), None)
  File "<string>", line 3, in raise_from
urllib3.exceptions.LocationParseError: Failed to parse: https://192.168.100.207:443https:/192.168.100.207/redfish/v1/SessionService/Sessions/admin0000000062b3f0f8b645a1cb/
```

The reason is that the session URl returned by iLO does not include the
port. Therefore the code in the session_location getter:

```
session_loc = self._session_location.replace(self.base_url, '')
```

fails to remove the base URL and we end up with a malfored URL as seen
above. The fix is to change the session URL based on the base URL.

-----Synopsis of Commits Above-----

**Please fill out the following when submitting the PR**

## Status
- [x] READY 
- [ ] IN-DEVELOPMENT 
- [ ] ON-HOLD

## Additional High Level Description
A few sentences describing the overall goals of the pull request's commits.
- Include the issue number here if it fixes one. Example `#1`
- Include the QUIX number here if it fixes one.

## Dependent PRs
List any PRs that must be merged together. (Tool dependent PRs SHOULD NOT occur.)
- If additional library PRs are to be referenced simply note the PR# 

## Before Status can be set to READY I have completed the following:
- [ ] Check if documentation updates
- [ ] Check if example code updates
- [ ] Pylint static analysis tests are passing above 9.0/10.0
- [ ] Unit tests added for any new functions/features and passed.
- [ ] Targeted/Custom Testing performed. Indicate the results in the description above or as additional comments.

### CODE REVIEW

**Comment by JohnAZoidberg** [comment]
I care about this because sometimes I use iLO with non-standard port.
Or I tunnel access to iLO to `https://localhost:4443`

File: src/redfish/rest/v1.py [hunk 1]
```diff
@@ -307,7 +307,21 @@ def session_location(self):
     @session_location.setter
     def session_location(self, ses_loc):
         """Set the session URI"""
-        self._session_location = ses_loc
+        from urllib.parse import urlparse
```
Comment (JohnAZoidberg) [inline src/redfish/rest/v1.py:310]: Probably should move the import up to the top.
And I'm not sure if you care about Python2 support? Hopefully not anymore in 2022. But I think the import for Python2 would be different.

File: src/redfish/rest/v1.py [hunk 2]
```diff
@@ -307,7 +307,21 @@ def session_location(self):
...
+        new_port = login_url_p.port
+        # The session_url returned by iLO does not include the port information
+        # If we're using a non-standard port to connect to iLO we have to
+        # manually insert it, based on the base_url.
+        if new_port and ses_loc:
+            print(ses_loc)
+            session_location_p = urlparse(ses_loc)
+            session_location_p = session_location_p._replace(netloc=f"{session_location_p.hostname}:{new_port}")
+            print(session_location_p)
+            print(session_location_p.geturl())
```
Comment (JohnAZoidberg) [inline src/redfish/rest/v1.py:321]:
```suggestion
            session_location_p = urlparse(ses_loc)
            session_location_p = session_location_p._replace(netloc=f"{session_location_p.hostname}:{new_port}")
```

File: src/redfish/rest/v1.py (hunk 1 above)
Comment (rajeevkallur) [reply src/redfish/rest/v1.py:310]: Yes, Python2 is not supported from version 3.2.0

**Comment by rajeevkallur** [comment]
This will be merged in the next release that is 3.6

**Pull request closed without merging**